src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -987,34 +987,33 @@ public static bool IsDefaultExpressionContext(this SyntaxTree syntaxTree, int po
             this SyntaxTree syntaxTree,
             int position,
             SyntaxToken tokenOnLeftOfPosition,
-            CancellationToken cancellationToken,
-            bool includeOperators = false,
-            bool isThisKeyword = false)
+            bool includeOperators,
+            out int parameterIndex,
+            out SyntaxKind previousModifier)
         {
             // cases:
             //   Goo(|
             //   Goo(int i, |
             //   Goo([Bar]|
-            var token = tokenOnLeftOfPosition;
-            token = token.GetPreviousTokenIfTouchingWord(position);
+            var token = tokenOnLeftOfPosition.GetPreviousTokenIfTouchingWord(position);
+
+            parameterIndex = -1;
+            previousModifier = SyntaxKind.None;
 
             if (token.IsKind(SyntaxKind.OpenParenToken) &&
                 token.Parent.IsDelegateOrConstructorOrLocalFunctionOrMethodOrOperatorParameterList(includeOperators))
             {
+                parameterIndex = 0;
                 return true;
             }
 
             if (token.IsKind(SyntaxKind.CommaToken) &&
                 token.Parent.IsDelegateOrConstructorOrLocalFunctionOrMethodOrOperatorParameterList(includeOperators))
             {
-                if (isThisKeyword)
-                {
-                    var parameterList = token.GetAncestor<ParameterListSyntax>();
-                    var commaIndex = parameterList.Parameters.GetWithSeparators().IndexOf(token);
-                    var index = commaIndex / 2 + 1;
-                    return index == 0;
-                }
+                var parameterList = token.GetAncestor<ParameterListSyntax>();
```

Comment:
GetAncestor seems unnecesary.  the check above let you know hte parent was a parameter list . (note, i realize this code existed like this before.  just pointing it out :))

Reply:
➡️ Fixed in e9a22f1b8615544e77515c71bab8f8af21752f5f